x-pack/plugin/esql/qa/testFixtures/src/main/resources/scoring.csv-spec

```diff
@@ -313,3 +313,43 @@ book_no:keyword | title:text
 4023            | A Tolkien Compass: Including J. R. R. Tolkien's Guide to the Names in The Lord of the Rings | foobar
 2714            | Return of the King Being the Third Part of The Lord of the Rings                            | foobar
 ;
+
+
+semanticTextMatch
+required_capability: metadata_score
+required_capability: semantic_text_type
+required_capability: match_function
+
+from semantic_text metadata _id, _score
+| where match(semantic_text_field, "something")
+| sort _score desc
+| keep _id, _score
+;
+
+_id:keyword | _score:double
+2           | 1.2879333961116942E19
+3           | 4.955684723851526E18
+1           | 1.9332616743917978E18
+;
+
+semanticTextMatchWithAllTheTextFunctions
+
+required_capability: metadata_score
+required_capability: semantic_text_type
+required_capability: match_function
+required_capability: kql_function
+required_capability: qstr_function
+
+from semantic_text metadata _id, _score
+| where match(semantic_text_field, "something")
+        AND match(description, "some")
+        AND kql("description:some*")
+        AND NOT qstr("host:host1")
+| sort _score desc
+| keep _id, _score
+;
+
+_id:keyword | _score:double
+2           | 1.2879333961116942E19
```

Comment:
Yes - I am not sure either - for now I followed the pattern we already have for testing with `_score`.
Happy to change this, but not as part of this PR.
I think it's a bit more complicated than that - if we could just make sure we use the same sharding strategy consistently in the scoring tests, this shouldn't be a problem anymore.
The sharding does not only affect the value of the scores, but as we have seen from some of the failing tests, it can also affect the order of the documents, with some document getting scored higher than other depending on the sharding strategy. So to fix this scoring tests, it is not sufficient to just check that the scores are greater than 0.

Reply:
> Happy to change this, but not as part of this PR.

++

> to fix this scoring tests, it is not sufficient to just check that the scores are greater than 0.

What worked for me to [test knn rescoring](https://github.com/elastic/elasticsearch/pull/116663/files#diff-6334703ba66256422f1712dce65de834f9a22a9456f68baeeb9a9c1afeaa97d9R580) is to get the results from an actual search and compare them directly to the ones returned from another, both scoring and the results themselves. We can check on some follow up on the strategy